lib/rets/metadata/root.rb

```diff
@@ -66,6 +66,10 @@ def marshal_dump
         sources
       end
 
+      def marshal_load(sources)
+        @sources = sources
+      end        
```

Comment:
If you remove it, the tests I didn't write would fail :).  That's there to make Marshal.load work.

Which reminds me, I need to add something to the README explaining that the cache object should use Marshal.dump and Marshal.load to persist/recover the metadata.